### PR TITLE
Try to reproduce #1749

### PR DESCRIPTION
**Pull request by forki**
This tries to reproduce the bug reported in #1749

### CODE REVIEW

File: tests/fsharpqa/Source/Conformance/InferenceProcedures/TypeInference/IgnoreUnitParameters.fs [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    
+    if a <> a' then
+        failwith "bug in a"
+    
+
+    
+    let run2 (f:System.Func<unit,string>) = f.Invoke()
+
+    let b = run2 (System.Func<unit,string>(fun () -> "hello world 2"))
+    let b' = run2 (System.Func<unit,string>(fun _ -> "hello world 2"))
```
Comment (KevinRansom) [inline tests/fsharpqa/Source/Conformance/InferenceProcedures/TypeInference/IgnoreUnitParameters.fs]: This is a scenario too.

let b'' = run2 (System.Func<unit,string>(fun x -> "hello world 2"))

File: tests/fsharpqa/Source/Conformance/InferenceProcedures/TypeInference/IgnoreUnitParameters.fs [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+    let d = run4 (System.Func<_>(fun () -> "hello world 4"))
+    let d' = run4 (System.Func<_>(fun _ -> "hello world 4"))
+    let d'' = run4 (System.Func<_>(fun x -> "hello world 4"))
+
+
+    if d <> d' || d' <> d'' then
+        failwith "bug in d"
+
+    failwith "all good"
```
Comment (forki) [inline tests/fsharpqa/Source/Conformance/InferenceProcedures/TypeInference/IgnoreUnitParameters.fs]: @KevinRansom @dsyme I put this into the code but the test is not red!?
I thought the code gets compiled and the executed? What do I need to change to let my code run?

**Comment by forki** [comment]
Ok this doesn't reproduce #1749, but at least we can use it to make sure we don't break similar cases

**Comment by dsyme** [comment]
@forki Thanks for adding the test case

**Pull request merged**